scalableminds/webknossos
# Turning off recording when leaving arbitrary modes

This PR automatically turns off recording when switching from arbitrary mode to orthogonal mode.
This solves the issue when having recording enabled and then switching to orthogonal, moving around and then switching back to any arbitrary view mode a new node is created instantly.
Now the user has to turn on the recording on changing back to any arbitrary mode.

### URL of deployed dev instance (used for testing):
- https://turnoffrecordinginorthogonal.webknossos.xyz

### Steps to test:
- Open a skeleton capable tracing and go to any arbitrary view mode. 
- Try out whether the bug still exists. A way to reproduce is (copied from issue #4164):
1. Set some nodes in flight mode. (Recording = True)
2. Switch to orthogonal mode.
3. Move viewport some nanometers away from the original point.
4. Switch back to orthogonal mode. --> a new node appears in the middle of the viewport.
If this procedure is done while flight mode is in watching mode (not recording), then no new nodes are created.

### Issues:
- fixes #4164

------
